src/vconsole/vconsole-setup.c

```diff
@@ -82,7 +84,8 @@ static bool is_settable(int fd) {
          *
          * http://lists.freedesktop.org/archives/systemd-devel/2013-February/008573.html
          */
-        return r == 0 && IN_SET(curr_mode, K_XLATE, K_UNICODE);
+        return r < 0 ? -errno :
+               IN_SET(curr_mode, K_XLATE, K_UNICODE) ? 0 : (errno = EBUSY, -EBUSY);
```

Comment:
errno is a dismal interface. We use errno when external interfaces give us on other choice (glibc is particularly bad in this regard), but when we some error code in errno, we convert it to local negative error value, and never use errno to pass state internally.